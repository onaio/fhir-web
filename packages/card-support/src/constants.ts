// OPENSRP API endpoints
export const OPENSRP_URL_USER_ASSIGNMENT = '/organization/user-assignment';
export const OPENSRP_URL_CLIENT_SEARCH = '/client/search';
export const OPENSRP_URL_LOCATION_HIERARCHY = '/location/hierarchy';

/** CSV File MIME Types */
export const TEXT_CSV = 'text/csv';
export const APPLICATION_CSV = 'application/csv';
export const TEXT_PLAIN = 'text/plain';
export const TEXT_X_CSV = 'text/x-csv';
export const APPLICATION_VND_EXCEL = 'application/vnd.ms-excel';
export const APPLICATION_X_CSV = 'application/x-csv';
export const TEXT_COMMA_SEPARATED_VALUES = 'text/comma-separated-values';
export const TEXT_X_COMMA_SEPARATED_VALUES = 'text/x-comma-separated-values';
export const TEXT_TAB_SEPARATED_VALUES = 'text/tab-separated-values';
