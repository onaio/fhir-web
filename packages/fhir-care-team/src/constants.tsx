// url

// Query patams
export const SEARCH_QUERY_PARAM = 'searchQuery';

// FHIR API strings
export const URL_ADMIN = '/admin';
export const FHIR_CARE_TEAM = 'CareTeam';
export const FHIR_GROUPS = 'Group';
export const FHIR_PRACTITIONERS = 'Practitioner';
export const URL_EDIT_CARE_TEAM = `${URL_ADMIN}/CareTeam/edit`;
export const URL_CREATE_CARE_TEAM = `${URL_ADMIN}/CareTeam/new`;
export const URL_CARE_TEAM = `${URL_ADMIN}/CareTeams`;

// Route params
export const ROUTE_PARAM_CARE_TEAM_ID = 'careTeamId';
