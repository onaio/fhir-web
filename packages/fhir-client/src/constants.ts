export const dateFormat = 'yyyy-MM-dd';
