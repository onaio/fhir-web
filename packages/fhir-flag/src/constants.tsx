// urls
export const URL_ADMIN = '/admin';
export const URL_CLOSE_FLAGS = `${URL_ADMIN}/close-flags`;

// // magic strings

export const thatiMinutes = 30 * 60 * 1000;

// form fields
export const locationName = 'locationName' as const;
export const productName = 'productName' as const;
export const status = 'status' as const;
export const comments = 'comments' as const;
