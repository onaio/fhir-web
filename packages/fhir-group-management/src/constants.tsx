// url
export const ADD_EDIT_GROUP_URL = '/group/add-edit';
export const ADD_EDIT_COMMODITY_URL = '/commodity/add-edit';
export const LIST_GROUP_URL = '/groups/list';
export const LIST_COMMODITY_URL = '/commodity/list';

// magic strings
export const groupResourceType = 'Group';
export const listResourceType = 'List';

// form constants
export const id = 'id' as const;
export const identifier = 'identifier' as const;
export const name = 'name' as const;
export const active = 'active' as const;
export const type = 'type' as const;
export const unitOfMeasure = 'unitOfMeasure' as const;
