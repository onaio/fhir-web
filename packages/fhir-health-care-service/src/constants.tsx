// url
export const URL_ADD_HEALTHCARE = 'healthcare/add/';
export const URL_EDIT_HEALTHCARE = 'healthcare/edit/';

//
// Healthcares
//
export const HEALTH_CARE_SERVICE_RESOURCE_TYPE = 'HealthcareService';
export const HEALTHCARES_ENDPOINT = 'HealthcareService/';
