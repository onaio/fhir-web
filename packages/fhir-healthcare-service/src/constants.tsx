// url
export const ADD_EDIT_HEALTHCARE_SERVICE_URL = '/healthcare/add-edit';
export const LIST_HEALTHCARE_URL = '/healthcare/list';

//
// constants
//
export const healthCareServiceResourceType = 'HealthcareService';

// form constants
export const id = 'id' as const;
export const identifier = 'identifier' as const;
export const name = 'name' as const;
export const active = 'active' as const;
export const comment = 'comment' as const;
export const extraDetails = 'extraDetails' as const;
export const providedBy = 'providedBy' as const;
