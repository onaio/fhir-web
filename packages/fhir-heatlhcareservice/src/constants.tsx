// url
export const URL_ADD_HEALTHCARE = 'healthcare/add/';
export const URL_EDIT_HEALTHCARE = 'healthcare/edit/';

//
// Healthcares
//
export const HEALTHCARE_UUID = 'healthcare-uuid';
export const HEALTHCARES_GET = 'HealthcareService/';
export const HEALTHCARES_PUT = 'HealthcareService/';
export const HEALTHCARES_POST = 'HealthcareService/';

//
// Healthcares
//
export const ORGANIZATION_GET = 'Organization/';
