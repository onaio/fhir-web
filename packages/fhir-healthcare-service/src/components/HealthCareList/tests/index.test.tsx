import { HealthCareList } from '..';
import React from 'react';
import { store } from '@opensrp/store';
import { createMemoryHistory } from 'history';
import { Route, Router, Switch } from 'react-router';
import { Provider } from 'react-redux';
import { authenticateUser } from '@onaio/session-reducer';
import { QueryClient, QueryClientProvider } from 'react-query';
import nock from 'nock';
import { waitForElementToBeRemoved } from '@testing-library/dom';
import { cleanup, fireEvent, render, screen, waitFor } from '@testing-library/react';
import {
  healthCareServicePage1,
  healthCareServicePage2,
  healthCareServiceSearch,
} from './fixtures';
import userEvents from '@testing-library/user-event';
import { healthCareServiceResourceType, LIST_HEALTHCARE_URL } from '../../../constants';

jest.mock('fhirclient', () => {
  return jest.requireActual('fhirclient/lib/entry/browser');
});

jest.mock('@opensrp/react-utils', () => {
  const actual = jest.requireActual('@opensrp/react-utils');

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const SearchForm = (props: any) => {
    const { onChangeHandler } = props;
    return (
      <div className="search-input-wrapper">
        <input onChange={onChangeHandler} data-testid="search-form"></input>
      </div>
    );
  };
  return {
    ...actual,
    SearchForm,
  };
});

nock.disableNetConnect();

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});

const props = {
  fhirBaseURL: 'http://test.server.org',
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const AppWrapper = (props: any) => {
  return (
    <Provider store={store}>
      <QueryClientProvider client={queryClient}>
        <Switch>
          <Route exact path={`${LIST_HEALTHCARE_URL}`}>
            {(routeProps) => <HealthCareList {...{ ...props, ...routeProps }} />}
          </Route>
          <Route exact path={`${LIST_HEALTHCARE_URL}/:id`}>
            {(routeProps) => <HealthCareList {...{ ...props, ...routeProps }} />}
          </Route>
        </Switch>
      </QueryClientProvider>
    </Provider>
  );
};

beforeAll(() => {
  store.dispatch(
    authenticateUser(
      true,
      {
        email: 'bob@example.com',
        name: 'Bobbie',
        username: 'RobertBaratheon',
      },
      { api_token: 'hunter2', oAuth2Data: { access_token: 'sometoken', state: 'abcde' } }
    )
  );
});

afterEach(() => {
  nock.cleanAll();
  cleanup();
});

afterAll(() => {
  nock.enableNetConnect();
});

test('renders correctly when listing resources', async () => {
  const history = createMemoryHistory();
  history.push(LIST_HEALTHCARE_URL);

  nock(props.fhirBaseURL)
    .get(`/${healthCareServiceResourceType}/_search`)
    .query({ _total: 'accurate', _getpagesoffset: 0, _count: 20 })
    .reply(200, healthCareServicePage1)
    .persist();

  nock(props.fhirBaseURL)
    .get(`/${healthCareServiceResourceType}/_search`)
    .query({ _total: 'accurate', _getpagesoffset: 20, _count: 20 })
    .reply(200, healthCareServicePage2);

  nock(props.fhirBaseURL)
    .get(`/${healthCareServiceResourceType}/_search`)
    .query({ _total: 'accurate', _getpagesoffset: 0, _count: 20, 'name:contains': 'testing' })
    .reply(200, healthCareServiceSearch);

  render(
    <Router history={history}>
      <AppWrapper {...props}></AppWrapper>
    </Router>
  );

  await waitForElementToBeRemoved(document.querySelector('.ant-spin'));

  expect(document.querySelector('.page-header')).toMatchSnapshot('Header title');

  document.querySelectorAll('tr').forEach((tr, idx) => {
    tr.querySelectorAll('td').forEach((td) => {
      expect(td).toMatchSnapshot(`table row ${idx} page 1`);
    });
  });

  fireEvent.click(screen.getByTitle('2'));

  const waitForSpinner = async () => {
    return await waitFor(() => {
      expect(document.querySelector('.ant-spin')).not.toBeInTheDocument();
    });
  };

  await waitForSpinner();
  expect(history.location.search).toEqual('?pageSize=20&page=2');

  document.querySelectorAll('tr').forEach((tr, idx) => {
    tr.querySelectorAll('td').forEach((td) => {
      expect(td).toMatchSnapshot(`table row ${idx} page 2`);
    });
  });

  // works with search as well.
  const searchForm = document.querySelector('[data-testid="search-form"]');
  userEvents.type(searchForm as Element, 'testing');

  expect(history.location.search).toEqual('?pageSize=20&page=1&search=testing');

  await waitForSpinner();
  document.querySelectorAll('tr').forEach((tr, idx) => {
    tr.querySelectorAll('td').forEach((td) => {
      expect(td).toMatchSnapshot(`Search ${idx} page 1`);
    });
  });

  // remove search.
  userEvents.clear(searchForm as Element);
  expect(history.location.search).toEqual('?pageSize=20&page=1');

  // view details
  nock(props.fhirBaseURL)
    .get(`/${healthCareServiceResourceType}/323`)
    .reply(200, healthCareServicePage1.entry[0].resource);

  // target the initial row view details
  const dropdown = document.querySelector('tbody tr:nth-child(1) [data-testid="action-dropdown"]');
  fireEvent.click(dropdown as Element);

  const viewDetailsLink = screen.getByText(/View Details/);
  expect(viewDetailsLink).toMatchInlineSnapshot(`
    <span>
      View Details
    </span>
  `);
  fireEvent.click(viewDetailsLink);
  expect(history.location.pathname).toEqual('/healthcare/list');
  expect(history.location.search).toEqual('?pageSize=20&page=1&viewDetails=323');

  await waitForSpinner();

  // close view details
  const closeButton = document.querySelector('[data-testid="close-button"]');
  fireEvent.click(closeButton as Element);

  expect(history.location.pathname).toEqual('/healthcare/list');

  expect(nock.isDone()).toBeTruthy();
});

test('responds as expected to errors', async () => {
  const history = createMemoryHistory();
  history.push(LIST_HEALTHCARE_URL);

  nock(props.fhirBaseURL)
    .get(`/${healthCareServiceResourceType}/_search`)
    .query({ _total: 'accurate', _getpagesoffset: 0, _count: 20 })
    .replyWithError('coughid');

  render(
    <Router history={history}>
      <AppWrapper debugKey="responds" {...props}></AppWrapper>
    </Router>
  );

  await waitForElementToBeRemoved(document.querySelector('.ant-spin'));

  expect(screen.getByText(/coughid/)).toBeInTheDocument();
});
