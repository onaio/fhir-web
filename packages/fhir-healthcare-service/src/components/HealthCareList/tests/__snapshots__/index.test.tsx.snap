// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when listing resources: Header title 1`] = `
<h4
  class="ant-typography page-header css-dev-only-do-not-override-k7429z"
>
  HealthCare service list
</h4>
`;

exports[`renders correctly when listing resources: Search 1 page 1 1`] = `
<td
  class="ant-table-cell"
>
  testing  Healthcare Jan27
</td>
`;

exports[`renders correctly when listing resources: Search 1 page 1 2`] = `
<td
  class="ant-table-cell"
>
  <div>
    Active
  </div>
</td>
`;

exports[`renders correctly when listing resources: Search 1 page 1 3`] = `
<td
  class="ant-table-cell"
>
  1/27/2022
</td>
`;

exports[`renders correctly when listing resources: Search 1 page 1 4`] = `
<td
  class="ant-table-cell"
>
  <span
    class="d-flex align-items-center"
  >
    <a
      class="m-0 p-1"
      href="/healthcare/add-edit/323"
    >
      Edit
    </a>
    <div
      class="ant-divider css-dev-only-do-not-override-k7429z ant-divider-vertical"
      role="separator"
    />
    <span
      aria-label="more"
      class="anticon anticon-more ant-dropdown-trigger more-options"
      data-testid="action-dropdown"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </span>
</td>
`;

exports[`renders correctly when listing resources: table row 1 page 1 1`] = `
<td
  class="ant-table-cell"
>
  testing  Healthcare Jan27
</td>
`;

exports[`renders correctly when listing resources: table row 1 page 1 2`] = `
<td
  class="ant-table-cell"
>
  <div>
    Active
  </div>
</td>
`;

exports[`renders correctly when listing resources: table row 1 page 1 3`] = `
<td
  class="ant-table-cell"
>
  1/27/2022
</td>
`;

exports[`renders correctly when listing resources: table row 1 page 1 4`] = `
<td
  class="ant-table-cell"
>
  <span
    class="d-flex align-items-center"
  >
    <a
      class="m-0 p-1"
      href="/healthcare/add-edit/323"
    >
      Edit
    </a>
    <div
      class="ant-divider css-dev-only-do-not-override-k7429z ant-divider-vertical"
      role="separator"
    />
    <span
      aria-label="more"
      class="anticon anticon-more ant-dropdown-trigger more-options"
      data-testid="action-dropdown"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </span>
</td>
`;

exports[`renders correctly when listing resources: table row 1 page 2 1`] = `
<td
  class="ant-table-cell"
>
  testing  Healthcare Jan27
</td>
`;

exports[`renders correctly when listing resources: table row 1 page 2 2`] = `
<td
  class="ant-table-cell"
>
  <div>
    Active
  </div>
</td>
`;

exports[`renders correctly when listing resources: table row 1 page 2 3`] = `
<td
  class="ant-table-cell"
>
  1/27/2022
</td>
`;

exports[`renders correctly when listing resources: table row 1 page 2 4`] = `
<td
  class="ant-table-cell"
>
  <span
    class="d-flex align-items-center"
  >
    <a
      class="m-0 p-1"
      href="/healthcare/add-edit/323"
    >
      Edit
    </a>
    <div
      class="ant-divider css-dev-only-do-not-override-k7429z ant-divider-vertical"
      role="separator"
    />
    <span
      aria-label="more"
      class="anticon anticon-more ant-dropdown-trigger more-options"
      data-testid="action-dropdown"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </span>
</td>
`;

exports[`renders correctly when listing resources: table row 2 page 1 1`] = `
<td
  class="ant-table-cell"
>
  ANC Service
</td>
`;

exports[`renders correctly when listing resources: table row 2 page 1 2`] = `
<td
  class="ant-table-cell"
>
  <div>
    Active
  </div>
</td>
`;

exports[`renders correctly when listing resources: table row 2 page 1 3`] = `
<td
  class="ant-table-cell"
>
  6/16/2021
</td>
`;

exports[`renders correctly when listing resources: table row 2 page 1 4`] = `
<td
  class="ant-table-cell"
>
  <span
    class="d-flex align-items-center"
  >
    <a
      class="m-0 p-1"
      href="/healthcare/add-edit/313"
    >
      Edit
    </a>
    <div
      class="ant-divider css-dev-only-do-not-override-k7429z ant-divider-vertical"
      role="separator"
    />
    <span
      aria-label="more"
      class="anticon anticon-more ant-dropdown-trigger more-options"
      data-testid="action-dropdown"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </span>
</td>
`;

exports[`renders correctly when listing resources: table row 2 page 2 1`] = `
<td
  class="ant-table-cell"
>
  ANC Service
</td>
`;

exports[`renders correctly when listing resources: table row 2 page 2 2`] = `
<td
  class="ant-table-cell"
>
  <div>
    Active
  </div>
</td>
`;

exports[`renders correctly when listing resources: table row 2 page 2 3`] = `
<td
  class="ant-table-cell"
>
  6/16/2021
</td>
`;

exports[`renders correctly when listing resources: table row 2 page 2 4`] = `
<td
  class="ant-table-cell"
>
  <span
    class="d-flex align-items-center"
  >
    <a
      class="m-0 p-1"
      href="/healthcare/add-edit/313"
    >
      Edit
    </a>
    <div
      class="ant-divider css-dev-only-do-not-override-k7429z ant-divider-vertical"
      role="separator"
    />
    <span
      aria-label="more"
      class="anticon anticon-more ant-dropdown-trigger more-options"
      data-testid="action-dropdown"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </span>
</td>
`;
